### Limpieza del dataset Titanic

In [1]:
# importar las librerias

import pandas as pd # type: ignore

In [2]:
# importar el dataset titanic

df_titanic = pd.read_csv('./data_raw/titanic-1.csv')

In [3]:
# visualizar los datos

df_titanic.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [4]:
# visualizar los tipos de datos

df_titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Survived                 887 non-null    int64  
 1   Pclass                   887 non-null    int64  
 2   Name                     887 non-null    object 
 3   Sex                      887 non-null    object 
 4   Age                      887 non-null    float64
 5   Siblings/Spouses Aboard  887 non-null    int64  
 6   Parents/Children Aboard  887 non-null    int64  
 7   Fare                     887 non-null    float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.6+ KB


In [5]:
# visualizar los valores perdidos

df_titanic.isna().sum()

Survived                   0
Pclass                     0
Name                       0
Sex                        0
Age                        0
Siblings/Spouses Aboard    0
Parents/Children Aboard    0
Fare                       0
dtype: int64

In [6]:
# observar la estadística descriptiva de las variables

df_titanic.describe(include='all')

Survived      Pclass                Name   Sex         Age  \
count   887.000000  887.000000                 887   887  887.000000   
unique         NaN         NaN                 887     2         NaN   
top            NaN         NaN  Mr. Patrick Dooley  male         NaN   
freq           NaN         NaN                   1   573         NaN   
mean      0.385569    2.305524                 NaN   NaN   29.471443   
std       0.487004    0.836662                 NaN   NaN   14.121908   
min       0.000000    1.000000                 NaN   NaN    0.420000   
25%       0.000000    2.000000                 NaN   NaN   20.250000   
50%       0.000000    3.000000                 NaN   NaN   28.000000   
75%       1.000000    3.000000                 NaN   NaN   38.000000   
max       1.000000    3.000000                 NaN   NaN   80.000000   

        Siblings/Spouses Aboard  Parents/Children Aboard       Fare  
count                887.000000               887.000000  887.00000  
unique                      NaN                      NaN        NaN  
top                         NaN                      NaN        NaN  
freq                        NaN                      NaN        NaN  
mean                   0.525366                 0.383315   32.30542  
std                    1.104669                 0.807466   49.78204  
min                    0.000000                 0.000000    0.00000  
25%                    0.000000                 0.000000    7.92500  
50%                    0.000000                 0.000000   14.45420  
75%                    1.000000                 0.000000   31.13750  
max                    8.000000                 6.000000  512.32920

### Análisis inicial

* Encontramos 8 columnas y 887 registros sin valores nulos o perdidos
* Vemos tres tipos de dato: float64, int64 y object
* Encontramos que todos los registros son únicos

Pueden cambiarse los nombres de las columnas para facilitar el manejo y cambiar los tipos de datos a las variables que no se usarán para cálculos o para el modelo que se quiera.

In [7]:
# cambiar los nombres de las columnas

df_titanic.rename(columns={"Survived":"survived","Pclass":"p_class","Name":"name","Sex":"sex","Age":"age",
                           "Siblings/Spouses Aboard":"siblings_spouses_aboard",
                           "Parents/Children Aboard":"parents_children_aboard","Fare":"fare"}, inplace=True)

In [8]:
df_titanic["age"].dtype

dtype('float64')

In [9]:
# cambiar los tipos de variables necesarios

for i in df_titanic.columns:
    if df_titanic[i].dtype == 'int64':
        df_titanic[i]=df_titanic[i].astype(str)
        


In [10]:
for i in df_titanic.columns:
    if df_titanic[i].dtype == 'float64':
        df_titanic[i]=df_titanic[i].astype(int)

In [11]:
df_titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   survived                 887 non-null    object
 1   p_class                  887 non-null    object
 2   name                     887 non-null    object
 3   sex                      887 non-null    object
 4   age                      887 non-null    int64 
 5   siblings_spouses_aboard  887 non-null    object
 6   parents_children_aboard  887 non-null    object
 7   fare                     887 non-null    int64 
dtypes: int64(2), object(6)
memory usage: 55.6+ KB


### Finalmente

Teniendo el dataset limpio podemos guardar el resultado procesado.

In [12]:
# guardar los datos procesados

df_titanic.to_csv("./data_processed/titanic_processed.csv")


